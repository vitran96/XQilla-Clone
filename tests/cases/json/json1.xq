xquery version "1.0";

let $json := '{
     "firstName": "John",
     "lastName": "Smith",
     "age": 25,
     "lala": true,
     "noggin": null,
     "char test": "\"\\\n\t\u0041",
     "address": 
     {
         "streetAddress": "21 2nd Street",
         "city": "New York",
         "state": "NY",
         "postalCode": "10021"
     },
     "phoneNumber": 
     [
         {
           "type": "home",
           "number": "212 555-1234"
         },
         {
           "type": "fax",
           "number": "646 555-4567"
         }
     ]
 }'
return (
  let $xml := xqilla:parse-json($json)
  return ($xml, text{"&#xA;"}, xqilla:serialize-json($xml))
)
