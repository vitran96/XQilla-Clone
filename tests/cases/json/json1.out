<json type="object"><pair name="firstName" type="string">John</pair><pair name="lastName" type="string">Smith</pair><pair name="age" type="number">25</pair><pair name="lala" type="boolean">true</pair><pair name="noggin" type="null"/><pair name="char test" type="string">"\
	A</pair><pair name="address" type="object"><pair name="streetAddress" type="string">21 2nd Street</pair><pair name="city" type="string">New York</pair><pair name="state" type="string">NY</pair><pair name="postalCode" type="string">10021</pair></pair><pair name="phoneNumber" type="array"><item type="object"><pair name="type" type="string">home</pair><pair name="number" type="string">212 555-1234</pair></item><item type="object"><pair name="type" type="string">fax</pair><pair name="number" type="string">646 555-4567</pair></item></pair></json>
{"firstName":"John","lastName":"Smith","age":25,"lala":true,"noggin":null,"char test":"\"\\\n\tA","address":{"streetAddress":"21 2nd Street","city":"New York","state":"NY","postalCode":"10021"},"phoneNumber":[{"type":"home","number":"212 555-1234"},{"type":"fax","number":"646 555-4567"}]}
